cmd/compute-domain-kubelet-plugin/driver.go

```diff
@@ -38,63 +40,64 @@ const ErrorRetryMaxTimeout = 45 * time.Second
 
 // permanentError defines an error indicating that it is permanent.
 // By default, every error will be retried up to ErrorRetryMaxTimeout.
-// Errors marked as permament will not be retried.
+// Errors marked as permanent will not be retried.
 type permanentError struct{ error }
 
 func isPermanentError(err error) bool {
 	return errors.As(err, &permanentError{})
 }
 
-var _ drapbv1.DRAPluginServer = &driver{}
-
 type driver struct {
 	sync.Mutex
 	client coreclientset.Interface
-	plugin kubeletplugin.DRAPlugin
+	helper *kubeletplugin.Helper
 	state  *DeviceState
 }
 
 func NewDriver(ctx context.Context, config *Config) (*driver, error) {
-	driver := &driver{
-		client: config.clientsets.Core,
-	}
-
 	state, err := NewDeviceState(ctx, config)
 	if err != nil {
 		return nil, err
 	}
-	driver.state = state
 
-	plugin, err := kubeletplugin.Start(
+	driver := &driver{
+		client: config.clientsets.Core,
+		state:  state,
+	}
+
+	driver.helper, err = kubeletplugin.Start(
 		ctx,
-		[]any{driver},
+		driver,
 		kubeletplugin.KubeClient(driver.client),
 		kubeletplugin.NodeName(config.flags.nodeName),
 		kubeletplugin.DriverName(DriverName),
-		kubeletplugin.RegistrarSocketPath(PluginRegistrationPath),
-		kubeletplugin.PluginSocketPath(DriverPluginSocketPath),
-		kubeletplugin.KubeletPluginSocketPath(DriverPluginSocketPath))
+	)
 	if err != nil {
 		return nil, err
 	}
-	driver.plugin = plugin
 
 	// Enumerate the set of ComputeDomain daemon devices and publish them
-	var resources kubeletplugin.Resources
+	var slice resourceslice.Slice
```

Comment:
Using just `slice` here makes this section more difficult to read and understand. Does using `deviceSlice` or something similar help this?

Reply:
done